### PR TITLE
Improvements 

### PR DESCRIPTION
**Pull request by Yanpas**
Changes in commit messages

### CODE REVIEW

**Comment by Yanpas** [comment]
@MaskRay Travis CI should have been triggered on PR, you need to enable it on the site

File: src/globalContext.ts [hunk 1]
```diff
@@ -1,34 +1,67 @@
...
 export class GlobalContext implements Disposable {
+  public readonly chan: OutputChannel;
   private _dispose: Disposable[] = [];
   private _server: ServerContext;
+  private _isRunning = false;
+  private _srvCwd: string;
   public constructor(
   ) {
-    this._server = new ServerContext();
+    this.chan = window.createOutputChannel("ccls");
```
Comment (MaskRay) [inline src/globalContext.ts]: Regarding double quote vs single quote, I have found this: https://stackoverflow.com/questions/25453864/what-is-the-standard-for-use-of-quotes-in-typescript

The typescript code guideline (double quote) is not a community code guideline. Single quotes are prevalent in other JavaScript communities thus single quotes seem to be more favorable.

File: src/statusBarIcon.ts [hunk 2]
```diff
@@ -40,10 +42,16 @@ export class StatusBarIconProvider implements Disposable {
     let info: CclsInfoResponse;
     try {
       info = await this.client.sendRequest<CclsInfoResponse>("$ccls/info");
+      this.wasError = false;
     } catch (e) {
+      if (this.wasError) {
```
Comment (MaskRay) [inline src/statusBarIcon.ts]: Does link require mandatory `{}` around single simple statements? If yes,, it is very unfortunate.

Comment (Yanpas) [reply src/statusBarIcon.ts]: If you maen `{}` then yes, there is such rule, but it can be disabled.

File: src/globalContext.ts (hunk 1 above)
Comment (Yanpas) [reply src/globalContext.ts]: Seems due to html nature. Will fix it in separate pr. Hope linter can forbid double quotes

File: src/statusBarIcon.ts (hunk 2 above)
Comment (Yanpas) [reply src/statusBarIcon.ts]: https://palantir.github.io/tslint/rules/curly/

File: src/globalContext.ts (hunk 1 above)
Comment (Yanpas) [reply src/globalContext.ts]: https://palantir.github.io/tslint/rules/quotemark/

**Comment by Yanpas** [comment]
Merge & release?

**Comment by MaskRay** [comment]
Merged with some tweaks.

**Pull request closed without merging**